### DISCUSSION TITLE
Depiction of the first media file of structure element (document type) in the presentation

### DISCUSSION THREAD

**Discussion by andre-hohmann** [kitodo/kitodo-production#5548] [Q&A] [answered]
In Kitodo.Production 2, all media files are assigned to the top structural element (document type) of a process (monograph, volume, ...). Accordingly, the first media file is displayed in the presentation when the document type is clicked in the structure tree. 

In Kitodo.Production 3, only media files that are not assigned to any other structure element are assigned to the top structure element (document type) of an process. Accordingly, the first "unstructured" media file is displayed in the presentation when the document type is clicked in the structure tree - not the first media files of the process. 

Is this behavior known and intended? 

##### Process of Kitodo.Production 3
Klick on "`Monographie Entwurf einer Bekanntmachung des Raths zu Dresden`" -> media file 81 is shown

<img src="https://user-images.githubusercontent.com/14835770/220080536-405b1620-6c40-472e-aec3-479897a2524e.gif" width="500">

##### Process of Kitodo.Production 2
Klick on "`Monographie Entwurf einer Bekanntmachung des Raths zu Dresden`" -> media file 1 is shown

<img src="https://user-images.githubusercontent.com/14835770/220080565-2c753662-84d6-4876-bf79-82a802039d78.gif" width="500">

**Comment by BartChris** [comment, marked as answer]
I have no knowledge on Presentation (and Kitodo 2), but i am not able to reproduce the behavior, that only media files assigned to no substructure element are associated with the top level element. All elements are always associated with the root element inside the structLink section.

Edit: What i noticed however is that the unstructured page - even if it comes last in the physical structmap - comes first in the structLink section. Could that be the issue? 

I looked at the XML of https://digital.slub-dresden.de/werkansicht/dlf/82234/2 and this situation does not appear there: PHYS_0002 is an unstructured page and only assigned to LOG_0000, but it is in the "correct" (no idea if the METS profile of the DFG says something about that) position in the StructLink section. So it appears as if Kitodo Prod 3 handles that different.

Edit2: I checked the guidelines, it seems like Kitodo 3 is violating the spec here:
"Die Reihenfolge der mets:smLink für jede logische Struktur muss der korrekten Anordnung der physischen Strukturelemente innerhalb dieser logischen Struktur entsprechen", 

"The sequence of mets:smLink for every logical structure has to correspond to the correct arrangement of the physical structural
elements within this logical structure"

https://dfg-viewer.de/fileadmin/groups/dfgviewer/METS-Anwendungsprofil_2.3.1.pdf, S. 16
https://dfg-viewer.de/fileadmin/groups/dfgviewer/METS_application_profile_2.3.1.pdf, p. 16

**Comment by BartChris** [comment]
I made an issue for that: https://github.com/kitodo/kitodo-production/issues/5554

**Comment by andre-hohmann** [comment]
@BartChris : Thanks a lot for the investigation and opening the issue! 

You are right, the problem lies within the _<mets:structLink>_. 
Thus we can state, that the described behavior is not intended and even violates the METS standard. 
That should be solved according to: 
- #5554. 

I will mark the discussion as answered, because my initial question was answered.
